### PR TITLE
For examples and tests- fixes warnings, updates packages, and updates…

### PR DESCRIPTION
**Pull request by rossgrambo**
## Why this PR?

Planning to release soon- it's a good opportunity to update packages, examples, and fix some linting errors.

## Visible changes

Some minor version bumps in test packages and examples.

### CODE REVIEW

File: examples/TargetingConsoleApp/Program.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@
 services.AddSingleton(configuration)
         .AddFeatureManagement();
 
-IUserRepository userRepository = new InMemoryUserRepository();
+InMemoryUserRepository userRepository = new InMemoryUserRepository();
```
Comment (zhiyuanliang-ms) [inline examples/TargetingConsoleApp/Program.cs:20]:
```suggestion
var userRepository = new InMemoryUserRepository();
```

File: examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs [hunk 2]
```diff
@@ -5,21 +5,15 @@
...
 namespace VariantAndTelemetryDemo.Pages
 {
-    public class IndexModel : PageModel
+    public class IndexModel(
+        IVariantFeatureManager featureManager,
+        TelemetryClient telemetry) : PageModel
     {
-        private readonly IVariantFeatureManager _featureManager;
-        private readonly TelemetryClient _telemetry;
+        private readonly IVariantFeatureManager _featureManager = featureManager ?? throw new ArgumentNullException(nameof(featureManager));
```
Comment (zhiyuanliang-ms) [inline examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs:12]: Is this change triggered for any linting rule? I prefer the previous pattern.

File: src/Microsoft.FeatureManagement/FeatureManager.cs [hunk 3]
```diff
@@ -699,9 +699,7 @@ private ValueTask<VariantDefinition> AssignVariantAsync(EvaluationEvent evaluati
 
         private void BindSettings(IFeatureFilterMetadata filter, FeatureFilterEvaluationContext context, int filterIndex)
         {
-            IFilterParametersBinder binder = filter as IFilterParametersBinder;
-
-            if (binder == null)
+            if (!(filter is IFilterParametersBinder binder))
```
Comment (zhiyuanliang-ms) [inline src/Microsoft.FeatureManagement/FeatureManager.cs:702]: Is this change caused by any linting rule?

**Review by zhiyuanliang-ms — APPROVED** [review]

File: examples/TargetingConsoleApp/Program.cs (hunk 1 above)
Comment (rossgrambo) [reply examples/TargetingConsoleApp/Program.cs:20]: Did we commit to this format with our linter? I believe others prefer this way, but personally, I don't like it. Haha.

Comment (jimmyca15) [reply examples/TargetingConsoleApp/Program.cs:20]: Both of these options are fine.

The only thing that's not fine is using var when the type isn't obvious in the assignment.

Comment (jimmyca15) [reply examples/TargetingConsoleApp/Program.cs:20]: For the sake of taking a direction and rolling with it, our code base heavily prefers the option of

`var userRepository = new InMemoryUserRepository();`

As the type repetition is unnecessary.

In the future I expect we'll shift to the latest language syntax for this scenario

`InMemoryUserRepository userRepository = new();`

File: src/Microsoft.FeatureManagement/FeatureManager.cs (hunk 3 above)
Comment (rossgrambo) [reply src/Microsoft.FeatureManagement/FeatureManager.cs:702]: Yeah this was shown to me by code analysis. I can't find the rule- even if I undo the change, it's no longer flagging it... 

The check right below this uses this format anyway, so we should be consistent anyway.

File: examples/TargetingConsoleApp/Program.cs (hunk 1 above)
Comment (rossgrambo) [reply examples/TargetingConsoleApp/Program.cs:20]: Sounds good, I'll do with `var` for now. I like the latest when we're able to make the switch.

File: examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs (hunk 2 above)
Comment (rossgrambo) [reply examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs:12]: Yeah this was triggered by primary constructor: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0290

Which I do think looks nicer- although it's a little less nice when we have ArgumentNullException checks

Comment (zhiyuanliang-ms) [reply examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs:12]: > although it's a little less nice when we have ArgumentNullException checks

Yes. And I do think the previous one looks nicer to me. hahaha 

I don't remember our current linting rule set which only followed default settings of IDE0055 will enforce this. If not, I prefer to not this rule in the example projects.

File: src/Microsoft.FeatureManagement/FeatureManager.cs (hunk 3 above)
Comment (zhiyuanliang-ms) [reply src/Microsoft.FeatureManagement/FeatureManager.cs:702]: > so we should be consistent anyway

Do you mean to revert this change or to keep it?

Comment (rossgrambo) [reply src/Microsoft.FeatureManagement/FeatureManager.cs:702]: Either keep this change- or change the following if statement to use the same style as this one.

File: examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs (hunk 2 above)
Comment (rossgrambo) [reply examples/VariantAndTelemetryDemo/Pages/Index.cshtml.cs:12]: I don't feel strongly either way, so I'll go ahead and revert it.

**Review by jimmyca15 — APPROVED** [review]

**Pull request merged**